nirc_ehr/resources/queries/study/housing.js

```diff
@@ -0,0 +1,32 @@
+require("ehr/triggers").initScript(this);
+
+EHR.Server.TriggerManager.registerHandlerForQuery(EHR.Server.TriggerManager.Events.BEFORE_INSERT, 'study', 'housing', function (helper, scriptErrors, row, oldRow) {
+
+    if (helper.isETL()) {
+        // format - From Location: UL Lafayette - NIRC; To Location: 28 - F2/CW - Rm 205 - B03;
+        if (row.locationTransferText) {
+            // replaceAll doesn't work
+            let locationTransferText = row.locationTransferText.replace(';',':').replace(';',':').split(':');
+            // [0] = From Location,
+            // [1] =  UL Lafayette - NIRC,
+            // [2] =  To Location,
+            // [3] = 28 - F2/CW - Rm 205 - B03
+            // translate [3] to locationId
+            LABKEY.Query.selectRows({
```

Comment:
How is perf on this ETL? If it's slow, we might want to cache the locations instead of querying for every row

Reply:
cached the query, it is faster now.